src/main/java/org/jabref/gui/mergeentries/BatchEntryMergeWithFetchedDataAction.java

```diff
@@ -0,0 +1,176 @@
+import org.jabref.logic.l10n.Localization;
+import org.jabref.logic.util.BackgroundTask;
+import org.jabref.logic.util.NotificationService;
+import org.jabref.logic.util.TaskExecutor;
+import org.jabref.model.entry.BibEntry;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Handles the merging of multiple bibliography entries with fetched data.
+ * This action performs background fetching and merging of entries while
+ * providing progress updates to the user.
+ */
+public class BatchEntryMergeWithFetchedDataAction extends SimpleCommand {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BatchEntryMergeWithFetchedDataAction.class);
+
+    private final Supplier<LibraryTab> tabSupplier;
+    private final GuiPreferences preferences;
+    private final NotificationService notificationService;
+    private final TaskExecutor taskExecutor;
+
+    public BatchEntryMergeWithFetchedDataAction(Supplier<LibraryTab> tabSupplier,
+                                                GuiPreferences preferences,
+                                                NotificationService notificationService,
+                                                StateManager stateManager,
+                                                TaskExecutor taskExecutor) {
+        this.tabSupplier = tabSupplier;
+        this.preferences = preferences;
+        this.notificationService = notificationService;
+        this.taskExecutor = taskExecutor;
+        this.executable.bind(ActionHelper.needsDatabase(stateManager));
+    }
+
+    @Override
+    public void execute() {
+        LibraryTab libraryTab = tabSupplier.get();
+        if (libraryTab == null) {
+            LOGGER.error("Cannot perform batch merge: no library is open.");
+            return;
+        }
+
+        List<BibEntry> libraryEntries = libraryTab.getDatabase().getEntries();
+        if (libraryEntries.isEmpty()) {
+            notificationService.notify(Localization.lang("empty library"));
+            return;
+        }
+
+        MergeContext context = new MergeContext(
+                libraryTab,
+                libraryEntries,
+                new MergingIdBasedFetcher(preferences.getImportFormatPreferences()),
+                notificationService
+        );
+
+        taskExecutor.execute(createBackgroundTask(context));
+    }
+
+    private static BackgroundTask<List<String>> createBackgroundTask(MergeContext context) {
+        BackgroundTask<List<String>> task = new BackgroundTask<>() {
+            @Override
+            public List<String> call() {
+                return processMergeEntries(context, this);
+            }
+        };
+
+        task.setTitle(Localization.lang("Fetching and merging entries"));
+        task.showToUser(true);
+
+        task.onSuccess(updatedEntries ->
+                handleSuccess(updatedEntries, context));
+
+        task.onFailure(ex ->
+                handleFailure(ex, context.notificationService()));
```

Comment:
You can chain these calls.

General comment - do not do now - read on -- You can also use `wrap` for a shorter syntax. See org.jabref.gui.preferences.preview.PreviewTabViewModel#setValues for an example.